scripts/variantstore/variant_annotations_table/GvsValidateVAT.wdl

```diff
@@ -35,191 +52,247 @@ workflow GvsValidateVat {
-                               SchemaNonzeroAcAn.name,
-                               SubpopulationMax.name,
-                               SubpopulationAlleleCount.name,
-                               SubpopulationAlleleNumber.name,
-                               DuplicateAnnotations.name,
-                               ClinvarSignificance.name,
-                               SchemaAAChangeAndExonNumberConsistent.name,
-                               SpotCheckForAAChangeAndExonNumberConsistency.name,
-                               CheckForNullColumns.name
-                               ],
-            validation_results = [
-                                 EnsureVatTableHasVariants.result,
-                                 SpotCheckForExpectedTranscripts.result,
-                                 SchemaOnlyOneRowPerNullTranscript.result,
-                                 SchemaNullTranscriptsExist.result,
-                                 SchemaNoNullRequiredFields.result,
-                                 SchemaPrimaryKey.result,
-                                 SchemaEnsemblTranscripts.result,
-                                 SchemaNonzeroAcAn.result,
-                                 SubpopulationMax.result,
-                                 SubpopulationAlleleCount.result,
-                                 SubpopulationAlleleNumber.result,
-                                 DuplicateAnnotations.result,
-                                 ClinvarSignificance.result,
-                                 SchemaAAChangeAndExonNumberConsistent.result,
-                                 SpotCheckForAAChangeAndExonNumberConsistency.result,
-                                 CheckForNullColumns.result
-                                 ],
-            cloud_sdk_docker = effective_cloud_sdk_docker,
+    # only check certain things if the callset is larger than 10,000 samples (a guess)
+    Boolean callset_is_small = GetNumSamplesLoaded.num_samples < 10000
+    if (!callset_is_small) {
+        call ClinvarSignificance {
+            input:
+                project_id = project_id,
+                fq_vat_table = fq_vat_table,
+                last_modified_timestamp = VatDateTime.last_modified_timestamp,
+                cloud_sdk_docker = effective_cloud_sdk_docker,
+        }
+
+        call SpotCheckForAAChangeAndExonNumberConsistency {
+            input:
+                project_id = project_id,
+                fq_vat_table = fq_vat_table,
+                last_modified_timestamp = VatDateTime.last_modified_timestamp,
+                cloud_sdk_docker = effective_cloud_sdk_docker,
+        }
+
+        call CheckForNullColumns {
+            input:
+                project_id = project_id,
+                fq_vat_table = fq_vat_table,
+                last_modified_timestamp = VatDateTime.last_modified_timestamp,
+                variants_docker = effective_variants_docker
+        }
+
```

Comment:
Is it possible to define the arrays like 'pf_flags' conditionally? LIke:
pf_flags = if (!callset_is_small) { [A, B, C]} else { [A, B, C, D] }
Then you wouldn't need different invocations of calls to GenerateFinalReport?

No worries if not possible.

Reply:
It probably is, but you would have to do it for the pf_flags, validation_names and validation_results arrays, so I don't think it would save much space (if any).